operate/client/src/App/ProcessInstance/BottomPanel/VariablePanel/index.tsx

```diff
@@ -55,8 +74,24 @@ const VariablePanel = observer(function VariablePanel() {
                 content: <InputOutputMappings type="Output" />,
                 onClick: variablesStore.stopPolling,
               },
+              ...(hasFlowNodeListeners
+                ? [
+                    {
+                      id: 'listeners',
+                      testId: 'listeners-tab-button',
+                      ...(listenersFailureCount && {
+                        labelIcon: <WarningFilled />,
+                      }),
+                      label: 'Listeners',
+                      content: <Listeners listeners={listeners} />,
+                      removePadding: true,
+                      onClick: variablesStore.stopPolling,
+                    },
+                  ]
+                : []),
             ]),
       ]}
+      key={`tabview-has-listeners-${hasFlowNodeListeners}`}
```

Comment:
@pedesen I used this key to remount the component between flow nodes with and without Listeners, so now when you select the Listeners tab and switch to a flow node that has no Listeners, the Variables tab gets selected again. When changing back and forth between flow nodes where both do or do not have listeners, it will work as it worked before, keeping the previously selected tab active.

Good tip, by the way!

Reply:
Nice to see it works! :+1: